File: compiler/bench/inline2.c
// expected: 700000000
// Chain of small functions - GCC inlines entire chain
#include <stdio.h>

static long f1(long n) { return n + 1; }
static long f2(long n) { return f1(n) + 1; }
static long f3(long n) { return f2(n) + 1; }
static long f4(long n) { return f3(n) + 1; }
static long f5(long n) { return f4(n) + 1; }
static long f6(long n) { return f5(n) + 1; }
static long f7(long n) { return f6(n) + 1; }

int main() {
    long x = 0;
    for (int i = 0; i < 100000000; i++) x = f7(x);
    printf("%ld\n", x);
    return 0;
}

File: compiler/bench/inline2.fs
\ expected: 700000000
\ Chain of small functions - GCC inlines entire chain

: f1 ( n -- n ) 1+ ;
: f2 ( n -- n ) f1 1+ ;
: f3 ( n -- n ) f2 1+ ;
: f4 ( n -- n ) f3 1+ ;
: f5 ( n -- n ) f4 1+ ;
: f6 ( n -- n ) f5 1+ ;
: f7 ( n -- n ) f6 1+ ;

: main
  0 100000000 0 do f7 loop . cr ;
